### ISSUE TITLE
Config for binary and multi class problem?

### ISSUE THREAD

**Issue by Lincoln93** [dreamquark-ai/tabnet#144]
Is it possible to use `TabNet` for binary and multi class output problem? What would be the proper evaluation metrics for each? For binary and multi-class, how we're defining `sigmoid` and `BCE` in the implementation?

**Comment by Optimox** [comment]
you do not need to define anything the implementation works for binary multi class and regression please refer to #140 for more information, feel free to reopen is this does not help.

You can also refer to the notebooks example in the repo that show classification and regression

**Comment by zjgbz** [comment]
> you do not need to define anything the implementation works for binary multi class and regression please refer to #140 for more information, feel free to reopen is this does not help.
> 
> You can also refer to the notebooks example in the repo that show classification and regression

Hi @Optimox , May I have your help to identify which default loss function was assigned for `TabNetMultiTaskClassifier`? I believed that `BCELoss` might be the default loss function for the multi- binary classification task. However, I found that
1. In the example in this repo, the loss function for `TabNetMultiTaskClassifier` is `loss_fn=[torch.nn.functional.cross_entropy]*NB_TASKS`. -- Does it mean that the default loss function is `[torch.nn.functional.cross_entropy]*NB_TASKS`?
2. I tried to specify `loss_fn = torch.nn.functional.binary_cross_entropy_with_logits`, but it always triggered the error

`Target size (torch.Size([1024])) must be the same as input size (torch.Size([1024, 2]))`

The dimension of my y_train and y_valid is [1024, 3] binary numpy array standing for 3 tasks. May I have your help to correctly manually assign BCELoss?

Thank you very much.

**Comment by Optimox** [comment]
@zjgbz 
Actually you'll need to stick to cross_entropy with TabNetMultiTaskClassifier. This is because the final activation function must be a softmax not a sigmoid.

For binary tasks, you'll end up with the same results using binary cross entropy or cross entropy with only 2 classes. So I would just advise you yo leave the default loss function.

**Comment by zjgbz** [comment]
> @zjgbz Actually you'll need to stick to cross_entropy with TabNetMultiTaskClassifier. This is because the final activation function must be a softmax not a sigmoid.
> 
> For binary tasks, you'll end up with the same results using binary cross entropy or cross entropy with only 2 classes. So I would just advise you to leave the default loss function.

Hi @Optimox,

Thank you very much. I reviewed the source codes and understand that when my multiple tasks are multiple binary classification with the same weights, your default loss function is the same as `[torch.nn.functional.cross_entropy]*NB_TASKS`, and the same as `BCELoss` with the same weights. I really appreciate your clear explanation.
